extended/src/test/resources/gexf/data.gexf

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<gexf xmlns="http://gexf.net/1.3" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://gexf.net/1.3 http://gexf.net/1.3/gexf.xsd" version="1.2">
+  <meta lastmodifieddate="2009-03-20">
+    <creator>Gephi.org</creator>
+    <description>A Web network</description>
+  </meta>
+  <graph defaultedgetype="directed">
+    <attributes class="node">
+      <attribute id="0" title="url" type="string"/>
+      <attribute id="1" title="indegree" type="float"/>
+      <attribute id="2" title="frog" type="boolean">
+        <default>true</default>
+      </attribute>
+    </attributes>
+    <nodes>
+      <node id="0" label="Gephi">
+        <attvalues>
+          <attvalue for="0" value="http://gephi.org"/>
+          <attvalue for="1" value="1"/>
+        </attvalues>
+      </node>
+       <node id="5" label="Gephi">
+          <attvalues>
+            <attvalue for="0" value="http://test.gephi.org"/>
+            <attvalue for="1" value="2"/>
+          </attvalues>
+        </node>
+      <node id="1" label="Webatlas">
+        <attvalues>
+          <attvalue for="0" value="http://webatlas.fr"/>
+          <attvalue for="1" value="2"/>
+        </attvalues>
+      </node>
+      <node id="2" label="RTGI">
+        <attvalues>
+          <attvalue for="0" value="http://rtgi.fr"/>
+          <attvalue for="1" value="1"/>
+        </attvalues>
+      </node>
+      <node id="3" label="BarabasiLab">
+        <attvalues>
+          <attvalue for="0" value="http://barabasilab.com"/>
+          <attvalue for="1" value="1"/>
+          <attvalue for="2" value="false"/>
```

Comment:
Se non ho capito male, concettualmente questo "false" (essendoci il `<attribute id="2" title="frog" type="boolean">`),
dovrebbe essere convertito in `false` (booleano).
Verifica se viene fatto (forse no, in quanto il graphml non ha test di questo tipo).

In caso non venga fatto, crea all'inizio del parse una `Mappa<ID, tipoDato>` (quindi sarà Map.of("0", "string", "1", "float", "2", "boolean") in questo esempio), 
che puoi chiamare `dataMap`.

Poi, in questo caso, quando trovi il `for="2"`,
ti recuperi il tipo di dato dalla dataMap, (altrimenti prendi "string" come default), e richiami il convertValue() che sta in ExtendedUtil.

Questo metodo andrebbe però cambiato all'inizio in questo modo, e mettendo gli altri case in minuscolo, tipo `case "float"` invece di `case "Float"`:
```
    private static Object convertValue(String value, String typeName) {
        typeName = typeName.toLowerCase();
        
        switch (typeName) {
            case "boolean":
                return Util.toBoolean(value);

// .... resto dello switch case

```

Poi verifica quali tipi di dato sono supportati da qui (sulla sezione "Data types"): https://github.com/gephi/gexf/releases/download/1.3/gexf-13-primer.pdf

Reply:
se modifico i case di convertValue non rischiamo di spaccare il toValidValue per i casi in maiuscolo?

Reply:
Ah, andrebbe cambiato anche `typeName.replace("Array", "")` in `typeName.replace("array", "")`

Reply:
Poi, se proprio proprio non è fattibile per qualche ragione, creane un'altro ad-hoc

Reply:
Purtroppo convertValue è privato e il default crea degli array in un modo in cui non ci interessa. Sto provando a creare un nuovo metodo e testare tutti i tipi lista che sono dichiarati nella documentazione